src/java.base/share/classes/sun/net/util/SocketExceptions.java

```diff
@@ -67,14 +68,10 @@ public static IOException of(IOException e, SocketAddress addr) {
     }
 
     private static IOException ofInet(IOException e, InetSocketAddress addr) {
-        int port = addr.getPort();
-        String host = addr.getHostString();
         StringBuilder sb = new StringBuilder();
         sb.append(e.getMessage());
         sb.append(": ");
-        sb.append(host);
-        sb.append(':');
-        sb.append(Integer.toString(port));
+        sb.append(addr.toString());
         String enhancedMsg = sb.toString();
```

Comment:
do you need extra "enhancedMsg" ? you can directly use "return create(e, sb.toString());"

Reply:
Yes, and can reduced further with String.join
```
return create(e, String.join(": ", e.getMessage(), addr.toString()));
```

Reply:
Cool, that's really short. I've updated the PR.

Reply:
Hello @schmelter-sap, The javadoc of `String.join` states that it can throw a `NullPointerException` if any of the elements in `null`. As far as I can see, the `IOException` that may get thrown, probably isn't guaranteed to have a exception message, in which case this will run into a `NullPointerException`. Perhaps we should guard this code against that possibility?

Reply:
Right now, if "enhanced socket exceptions" is enabled, then it only be used for socket/network exceptions. These will have some exception messages but you are right that catering for this possibility would be good, to avoid needing to change it again in the future.

Reply:
> Hello @schmelter-sap, The javadoc of `String.join` states that it can throw a `NullPointerException` if any of the elements in `null`. As far as I can see, the `IOException` that may get thrown, probably isn't guaranteed to have a exception message, in which case this will run into a `NullPointerException`. Perhaps we should guard this code against that possibility?

On JDK 19: 

```
jshell> String.join(": ", null, "bar")
$3 ==> "null: bar"

```

Reply:
@dfuch, you are right - it looks like the implementation of that API contradicts the javadoc. So I think it might end up being a CSR to update the javadoc to match the current implementation of `String.join()`. Given this, plus what Alan stated about the exception message being non-null in the context of this code, I think what Ralf currently has in this PR, is OK.

Reply:
No the API is indeed correct, it will throw if either the delimiter or the vararg array is null :-)

Reply:
> No the API documentation is indeed correct, it will throw if either the delimiter or the vararg array is null :-)

Ah! I see the distinction now :)